vendor/brave-ios/Ledger/Data/BATLedgerDatabase.mm

```diff
@@ -40,1108 +26,368 @@ + (nullable NSString *)migrateCoreDataToSQLTransaction
+          SQLString(obj.faviconURL), SQLString(obj.url), SQLString(obj.provider)];
+}
+
++ (NSString *)recurringDonationInsertFor:(RecurringDonation *)obj
 {
-  return [self firstOfClass:UnblindedToken.class
-                 predicates:@[[NSPredicate predicateWithFormat:@"tokenID == %lld", tokenID]]
-            sortDescriptors:@[]
-                    context:context];
+  const auto recurringDonationInsert =
+  @"INSERT INTO \"recurring_donation\" "
+  "(publisher_id, amount, added_date) VALUES ("
+  "%@,"  // publisher_id LONGVARCHAR NOT NULL PRIMARY KEY UNIQUE
+  "%f,"  // amount DOUBLE DEFAULT 0 NOT NULL
+  "%lld" // added_date INTEGER DEFAULT 0 NOT NULL
+  ");";
+  return [NSString stringWithFormat:recurringDonationInsert,
+          SQLString(obj.publisherID), obj.amount, obj.addedDate];
 }
 
-+ (BATPromotion *)promotiomFromDBPromotion:(Promotion *)dbPromo context:(NSManagedObjectContext *)context
++ (NSString *)unblindedTokenInsertFor:(UnblindedToken *)obj
 {
-  const auto promotion = [[BATPromotion alloc] init];
-  promotion.id = dbPromo.promotionID;
-  promotion.version = dbPromo.version;
-  promotion.type = static_cast<BATPromotionType>(dbPromo.type);
-  promotion.publicKeys = dbPromo.publicKeys;
-  promotion.suggestions = dbPromo.suggestions;
-  promotion.approximateValue = dbPromo.approximateValue;
-  promotion.status = static_cast<BATPromotionStatus>(dbPromo.status);
-  promotion.expiresAt = [dbPromo.expiryDate timeIntervalSince1970];
-  promotion.credentials = ^BATPromotionCreds * _Nullable {
-    const auto dbCreds = [self getPromoCredsWithPromotionID:dbPromo.promotionID context:context];
-    if (!dbCreds) {
-      return nil;
+  const auto unblindedTokenInsert =
+  @"INSERT INTO \"unblinded_tokens\" "
+  "(token_id, token_value, public_key, value, promotion_id) VALUES ("
+  "%lld," // token_id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL
+  "%@,"   // token_value TEXT
+  "%@,"   // public_key TEXT
+  "%f,"   // value DOUBLE NOT NULL DEFAULT 0
+  "%@"    // promotion_id TEXT
+  ");";
+  return [NSString stringWithFormat:unblindedTokenInsert,
+          obj.tokenID, SQLNullableString(obj.tokenValue),
+          SQLNullableString(obj.publicKey), obj.value,
+          SQLNullableString(obj.promotionID)];
+}
+
++ (void)deleteCoreDataServerPublisherList:(nullable void (^)(NSError * _Nullable error))completion
+{
+  const auto context = [DataController newBackgroundContext];
+  
+  BLOG(ledger::LogLevel::LOG_INFO) << "CoreData: Deleting publisher list" << std::endl;
+  [context performBlock:^{
+    const auto fetchRequest = ServerPublisherInfo.fetchRequest;
+    fetchRequest.entity = [NSEntityDescription entityForName:NSStringFromClass(ServerPublisherInfo.class)
+                                      inManagedObjectContext:context];
+    NSError *error = nil;
+    const auto deleteRequest = [[NSBatchDeleteRequest alloc] initWithFetchRequest:fetchRequest];
+    [context executeRequest:deleteRequest error:&error];
```

Comment:
After this executes, do we still use core data code anywhere?

I'm asking because NSBatchDeleteRequest deletes directly from the persistent store, 
so if there are any other managed contexts that rely on this code, they will not be notified, unless you do it manually

Reply:
We have no more CoreData code used anywhere except:
- Fetching each model's (_except the ServerPublisher ones_) full contents for creating the migration script
- Batch deleting the server publisher list